opengrok-indexer/src/main/java/org/opengrok/indexer/history/Annotation.java

```diff
@@ -190,6 +191,53 @@ public int getFileVersionsCount() {
         return fileVersions.size();
     }
 
+    /**
+     * Generate the color palette for the annotated revisions.
+     * <p>
+     * First, take into account revisions which are tracked in history fields
+     * and compute their color. Secondly, use all other revisions in order
+     * which is undefined and generate the rest of the colors for them.
+     *
+     * @return map of (revision, css color string) for each revision in {@code getRevisions()}
+     * @see #getRevisions()
+     */
+    public Map<String, String> getColors() {
+        Map<String, String> colors = new HashMap<>();
+        boolean even = true;
+        int hue = 60;
+        final List<String> revisions =
+                getRevisions()
```

Comment:
What if you display annotation for older revision of the file ? Then the set of revisions would be smaller.

Reply:
Yes exactly.

Reply:
Colors are not preserved for the revision, the palette will be different for older revision, starting from yellow with bigger steps. I don't see it as a problem.

Reply:
Maybe it's time to change it to red :-) The old and new schemes are so different it does not matter with what color to start. Or maybe better yet: make it consistent with IDEA behavior.

Reply:
Okay, as long as getRevisions() does the right thing..

Would be nice to have this covered by tests.